internal/provider/resource_gitlab_group_access_token_test.go

```diff
@@ -0,0 +1,307 @@
+
+	rInt := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: providerFactories,
+		CheckDestroy:      testAccCheckGitlabGroupAccessTokenDestroy,
+		Steps: []resource.TestStep{
+			// Create a Group and a Group Access Token
+			{
+				Config: testAccGitlabGroupAccessTokenConfig(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabGroupAccessTokenExists("gitlab_group_access_token.this", &gat),
+					testAccCheckGitlabGroupAccessTokenAttributes(&gat, &testAccGitlabGroupAccessTokenExpectedAttributes{
+						name:        "my group token",
+						scopes:      map[string]bool{"read_repository": true, "api": true, "write_repository": true, "read_api": true},
+						expiresAt:   "2099-01-01",
+						accessLevel: gitlab.AccessLevelValue(gitlab.DeveloperPermissions),
+					}),
+				),
+			},
+			// Update the Group Access Token to change the parameters
+			{
+				Config: testAccGitlabGroupAccessTokenUpdateConfig(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabGroupAccessTokenExists("gitlab_group_access_token.this", &gat),
+					testAccCheckGitlabGroupAccessTokenAttributes(&gat, &testAccGitlabGroupAccessTokenExpectedAttributes{
+						name:        "my new group token",
+						scopes:      map[string]bool{"read_repository": false, "api": true, "write_repository": false, "read_api": false},
+						expiresAt:   "2099-05-01",
+						accessLevel: gitlab.AccessLevelValue(gitlab.MaintainerPermissions),
+					}),
+				),
+			},
+			// Add a CICD variable with Group Access Token value
+			{
+				Config: testAccGitlabGroupAccessTokenUpdateConfigWithCICDvar(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabGroupAccessTokenExists("gitlab_group_access_token.this", &gat),
+					testAccCheckGitlabGroupVariableExists("gitlab_group_variable.var", &groupVariable),
+					testAccCheckGitlabGroupAccessTokenAttributes(&gat, &testAccGitlabGroupAccessTokenExpectedAttributes{
+						name:        "my new group token",
+						scopes:      map[string]bool{"read_repository": false, "api": true, "write_repository": false, "read_api": false},
+						expiresAt:   "2099-05-01",
+						accessLevel: gitlab.AccessLevelValue(gitlab.MaintainerPermissions),
+					}),
+				),
+			},
+			//Restore Group Access Token initial parameters
+			{
+				Config: testAccGitlabGroupAccessTokenConfig(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabGroupAccessTokenExists("gitlab_group_access_token.this", &gat),
+					testAccCheckGitlabGroupAccessTokenAttributes(&gat, &testAccGitlabGroupAccessTokenExpectedAttributes{
+						name:        "my group token",
+						scopes:      map[string]bool{"read_repository": true, "api": true, "write_repository": true, "read_api": true},
+						expiresAt:   "2099-01-01",
+						accessLevel: gitlab.AccessLevelValue(gitlab.DeveloperPermissions),
+					}),
+				),
+			},
+			//Destroy Group Access Token
```

Comment:
Repeating myself from https://github.com/gitlabhq/terraform-provider-gitlab/pull/858#discussion_r804112877

Would you rather test destroys this way, or with `CheckDestroy` and creating the group using the GitLab client outside of the TestSteps?

Reply:
sounds good 👍